include/PrimeTower.h

```diff
@@ -40,13 +41,15 @@ class PrimeTower
     const unsigned int number_of_prime_tower_start_locations = 21; //!< The required size of \ref PrimeTower::wipe_locations
 
     std::vector<ExtrusionMoves> pattern_per_extruder; //!< For each extruder the pattern to print on all layers of the prime tower.
-    std::vector<ExtrusionMoves> pattern_per_extruder_layer0; //!< For each extruder the pattern to print on the first layer
+    std::vector<std::vector<ExtrusionMoves>> pattern_extra_brim_per_layer; //!< For each layer of each extruder, the extra pattern to be added for adhesion and/or strength
```

Comment:
Is one of the vectors over the extruders and the other over a number of patterns? Maybe typedef those to show the intent. It's not clear what the outer and inner indexes will be.

Reply:
Yes indeed, technically I should have called it  `pattern_extra_brim_per_layer_per_extruder` but that would have been even more confusing :laughing: Using typedefs is a good idea, I'll see what I can do.

Reply:
@scottsha Tell me if the new version makes it better